src/main/java/io/vertx/core/shareddata/AsyncMap.java

```diff
@@ -167,6 +167,23 @@ default void replace(K k, V v, Handler<AsyncResult<@Nullable V>> resultHandler)
    */
   Future<@Nullable V> replace(K k, V v);
 
+  /**
+   * Replace the entry only if it is currently mapped to some value
+   *
+   * @param k  the key
+   * @param v  the new value
+   * @param ttl  The time to live (in ms) for the entry
+   * @param resultHandler  the result handler will be passed the previous value
+   */
+  default void replace(K k, V v, long ttl, Handler<AsyncResult<@Nullable V>> resultHandler) {
+    replace(k, v, ttl).onComplete(resultHandler);
+  }
+
+  /**
+   * Same as {@link #replace(K, V, long, Handler)} but returns a {@code Future} of the asynchronous result
+   */
+  Future<@Nullable V> replace(K k, V v, long ttl);
```

Comment:
This is a new API method, we should provide a default impl that returns a `Future` failed with `UnsupportedOperationException`.

Reply:
Hi @tsegismont 
What about https://github.com/vert-x3/vertx-hazelcast/pull/132 ?
Can we do something similar with default method?

Reply:
Hi, the purpose of providing such a default implementation is to avoid a breaking change when releasing. In the other PR, we're talking about a feature that, as far as we know, is only supported in Hazelcast.

So we can't do something similar, it's a different matter.